cardano-testnet/src/Testnet/Property/Run.hs

```diff
@@ -60,17 +64,29 @@ runTestnet tn = do
       IO.exitFailure
 
 
-testnetProperty :: (Conf -> H.Integration ()) -> H.Property
-testnetProperty tn = integrationWorkspace "testnet" $ \workspaceDir -> do
-  conf <- mkConf workspaceDir
-
-  -- Fork a thread to keep alive indefinitely any resources allocated by testnet.
-  void . H.evalM . liftResourceT . resourceForkIO . forever . liftIO $ IO.threadDelay 10000000
-
-  void $ tn conf
-
-  H.failure -- Intentional failure to force failure report
-
+testnetProperty :: CardanoTestnetOptions -> (Conf -> H.Integration ()) -> H.Property
+testnetProperty CardanoTestnetOptions{cardanoOutputDir} tn =
+  case cardanoOutputDir of
+      Nothing -> do
+        integrationWorkspace "testnet" $ \workspaceDir -> do
+          mkConf workspaceDir >>= go
+      Just userOutputDir ->
+        integration $ do
+          absUserOutputDir <- H.evalIO $ makeAbsolute userOutputDir
+          dirExists <- H.evalIO $ doesDirectoryExist absUserOutputDir
+          (if dirExists then
+            -- Likely dangerous, but who are we to judge the user?
+            H.note_ $ "Reusing " <> absUserOutputDir
+          else do
+            liftIO $ createDirectory absUserOutputDir
+            H.note_ $ "Created " <> absUserOutputDir)
+          go $ Conf $ TmpAbsolutePath absUserOutputDir
```

Comment:
Actually, you should use `mkConf` here as well. It's the same at this moment, but it could lead to a sneaky bug when someone adds more defaults to it.
```suggestion
          go $ mkConf absUserOutputDir
```

I think the whole `case` should just return the directory, and there should be a single call to `mkConf`.